src/app/container/paramfiles/paramfiles.component.ts

```diff
@@ -42,30 +42,59 @@ export class ParamfilesComponent extends EntryFileSelector {
   public entryType = 'tool';
   public downloadFilePath: string;
 
-  constructor(private containerService: ContainerService, private containersService: ContainersService,
-              private paramfilesService: ParamfilesService,
+  constructor(private containerService: ContainerService, private containersService: ContainersService, private gA4GHService: GA4GHService,
+              private paramfilesService: ParamfilesService, private gA4GHFilesStateService: GA4GHFilesStateService,
               public fileService: FileService) {
-    super();
+    super(fileService);
       this.published$ = this.containerService.toolIsPublished$;
   }
   getDescriptors(version): Array<any> {
     return this.paramfilesService.getDescriptors(this._selectedVersion);
   }
 
-  getFiles(descriptor): Observable<any> {
-    return this.paramfilesService.getFiles(this.id, 'containers', this._selectedVersion.name, this.currentDescriptor);
+  /**
+   * Get all the language-specific test parameter files
+   *
+   * @param {(('cwl' | 'wdl' | 'nfl'))} descriptor The descriptor language selected
+   * @returns {Observable<Array<ToolFile>>} The array of language-specific test parameter files
+   * @memberof ParamfilesComponent
+   */
+  getFiles(descriptor: ('cwl' | 'wdl' | 'nfl')): Observable<Array<ToolFile>> {
```

Comment:
You could define a string literal type for this. Just saying because I see it repeated. Hmm, I see you have a DescriptorType below, so maybe there's a reason you have to do it this way.

Reply:
DescriptorType enum is happen to be full caps